.github/workflows/linux-qe-template.yml

```diff
@@ -0,0 +1,200 @@
+            -e DEBUG=true \
+            -e SSH_AUTH_SOCK=$(cat ssh_auth_sock) \
+            -v "$(cat ssh_auth_sock):$(cat ssh_auth_sock)" \
+            -v $PWD/pull-secret:/opt/crc/pull-secret:z \
+            -v $PWD:/data:z \
+            quay.io/crcont/crc-${{inputs.qe-type}}:gh-linux-arm64 \
+                ${cmd} 
+          podman logs -f crc-${{ env.commit_sha }}-${{inputs.qe-type}}-${{inputs.preset}}
+
+      - name: Test Report
+        working-directory: ${{ env.commit_sha }}-${{inputs.qe-type}}-${{inputs.preset}}
+        id: test-report
+        uses: mikepenz/action-junit-report@v4
+        if: always() 
+        with:
+          fail_on_failure: true
+          include_passed: true
+          detailed_summary: true
+          require_tests:  true
+          report_paths: '**/*.xml'
+      
+      - name: Upload e2e results
+        working-directory: ${{ env.commit_sha }}-${{inputs.qe-type}}-${{inputs.preset}}
+        uses: actions/upload-artifact@v4
+        if: always()
+        with:
+          name: linux-${{inputs.qe-type}}-${{inputs.preset}}
+          path: |
+            **/*.xml
+            **/*.results
+            **/*.log
+
+      - name: Update status of the PR check
+        if: always()
+        run: |
+          set -xuo
+          # Status msg
+          data="{\"state\":\"success\""
+          if [[ ${{steps.test-report.outcome}} != "success" ]]; then 
+            data="{\"state\":\"failure\""
+          fi
+          data="${data},\"description\":\"Finished ${{inputs.qe-type}}-${{inputs.preset}} on Linux\""
+          data="${data},\"context\":\"${{ env.status_context }}\""
+          data="${data},\"target_url\":\"https://github.com/${{ github.repository }}/actions/runs/${{ github.run_id }}\"}"
+          # Create status by API call
+          curl -L -v -X POST \
+            -H "Accept: application/vnd.github+json" \
+            -H "Authorization: Bearer ${{ github.token }}" \
+            https://api.github.com/repos/${{ github.repository }}/statuses/${{ env.commit_sha }} \
+            -d "${data}"
+
+      - name: Env clear
+        working-directory: ${{ env.commit_sha }}-${{inputs.qe-type}}-${{inputs.preset}}
+        run: |
+          podman run --rm -d --privileged --name crc-linux-return-${{ env.commit_sha }}-${{inputs.qe-type}}-${{inputs.preset}} \
+            -e TARGET_HOST=$(cat host) \
+            -e TARGET_HOST_USERNAME=$(cat username) \
+            -e TARGET_HOST_KEY_PATH=/data/id_rsa \
+            -e BASTION_HOST_USERNAME=$(cat bastion_username) \
+            -e BASTION_HOST=$(cat bastion_host) \
+            -e TARGET_FOLDER=crc-support \
+            -e TARGET_CLEANUP='false' \
+            -e OUTPUT_FOLDER=/data \
+            -e DEBUG='true' \
+            -e SSH_AUTH_SOCK=$(cat ssh_auth_sock) \
+            -v "$(cat ssh_auth_sock):$(cat ssh_auth_sock)" \
+            -v ${PWD}:/data:z \
+            quay.io/rhqp/crc-support:v0.5.1-linux return2testingfarm
```

Comment:
What is this?

This is how the target is decommissioned? so you have to run it from inside the target :thinking: 

Tough if this is the reason Can you change the name of the step to make it clear? Or consider adding some comments on it to help others out to better understand the step.